pkg/github/github.go

```diff
@@ -127,13 +127,31 @@ type Client interface {
 	) (*github.IssueComment, *github.Response, error)
 }
 
+// TODO: we should clean up the functions listed below and agree on the same
+// return type (with or without error):
+// - New
+// - NewWithToken
+// - NewEnterprise
+// - NewEnterpriseWithToken
+
 // New creates a new default GitHub client. Tokens set via the $GITHUB_TOKEN
 // environment variable will result in an authenticated client.
 // If the $GITHUB_TOKEN is not set, then the client will do unauthenticated
 // GitHub requests.
 func New() *GitHub {
-	ctx := context.Background()
 	token := util.EnvDefault(TokenEnvKey, "")
+	client, err := NewWithToken(token)
+	if err != nil {
+		return nil
+	}
```

Comment:
yeah this doesn't really make sense considering NewWithToken never really returns `error` -- doing it to make `errcheck` linter happy.

Reply:
maybe we ignore that and add a nolint comment?

```
client, _ := NewWithToken(token) // nolint: errcheck
```

Reply:
sure, I'm not too opinionated on either approaches 😄 I'll switch it to nolint if nobody says anything today

Reply:
eh whatever I'll do it now lol